pkg/logs/adapter/minio_test.go

```diff
@@ -0,0 +1,194 @@
+
+func init() {
+	rand.New(rand.NewSource(time.Now().UnixNano()))
+}
+
+var letterRunes = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+func RandString(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = letterRunes[rand.Intn(len(letterRunes))]
+	}
+	return string(b)
+}
+
+func TestLogs(t *testing.T) {
+	t.Skip("skipping test")
+	consumer := NewMinioConsumer("localhost:9000", "minio", "minio123", "", "", "test-1", minioconnecter.Insecure())
+	id := "test-bla"
+	for i := 0; i < 1000; i++ {
+		fmt.Println("sending", i)
+		consumer.Notify(id, events.Log{Time: time.Now(),
+			Content: fmt.Sprintf("Test %d: %s", i, hugeString),
+			Type:    "test", Source: strconv.Itoa(i)})
+		time.Sleep(100 * time.Millisecond)
+	}
+	err := consumer.Stop(id)
+	assert.NoError(t, err)
+}
+
+func BenchmarkLogs(b *testing.B) {
+	randomString := RandString(5)
+	bucket := "test-bench"
+	consumer := NewMinioConsumer("localhost:9000", "minio", "minio123", "", "", bucket, minioconnecter.Insecure())
+	id := "test-bench" + "-" + randomString + "-" + strconv.Itoa(b.N)
+	totalSize := 0
+	for i := 0; i < b.N; i++ {
+		consumer.Notify(id, events.Log{Time: time.Now(),
+			Content: fmt.Sprintf("Test %d: %s", i, hugeString),
+			Type:    "test", Source: strconv.Itoa(i)})
+		totalSize += len(hugeString)
+	}
+	sizeInMB := float64(totalSize) / 1024 / 1024
+	err := consumer.Stop(id)
+	assert.NoError(b, err)
+	b.Logf("Total size for %s logs is %f MB", id, sizeInMB)
+}
+
+func BenchmarkLogs2(b *testing.B) {
+	//b.SetParallelism(1)
+	bucket := "test-bench"
+	consumer := NewMinioConsumer("localhost:9000", "minio", "minio123", "", "", bucket, minioconnecter.Insecure())
+	idChan := make(chan string, 100)
+	go verifyConsumer(idChan, bucket, consumer.minioClient)
+	var counter atomic.Int32
+	var wg sync.WaitGroup
+	for i := 0; i < 10; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			//randomString := RandString(5)
+			randomString := strconv.Itoa(int(counter.Add(1)))
+			id := "test-bench" + "-" + randomString
+			testOneConsumer(consumer, id)
+			idChan <- id
+		}()
+	}
+	wg.Wait()
+}
+
+func testOneConsumer(consumer *MinioConsumer, id string) {
+	fmt.Println("#####starting", id)
+	totalSize := 0
+	numberOFLogs := rand.Intn(100000)
+	for i := 0; i < numberOFLogs; i++ {
+		consumer.Notify(id, events.Log{Time: time.Now(),
+			Content: fmt.Sprintf("Test %d: %s", i, hugeString),
+			Type:    "test", Source: strconv.Itoa(i)})
+		totalSize += len(hugeString)
+		time.Sleep(time.Duration(rand.Intn(10)) * time.Millisecond)
+	}
+	sizeInMB := float64(totalSize) / 1024 / 1024
+	err := consumer.Stop(id)
+	if err != nil {
+		fmt.Println("#####error stopping", err)
+	}
+	//assert.NoError(b, err)
+	fmt.Printf("#####Total size for %s logs is %f MB\n\n\n", id, sizeInMB)
```

Comment:
what is a purpose of printing in unit tests?

Reply:
it is a benchmark func that it is run in benchmarking conditions and it provides more information about it